enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -101,12 +101,14 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
         String version = requestContext.getMatchedAPI().getAPIConfig().getVersion();
         context = context + "/" + version;
         ResourceConfig matchingResource = requestContext.getMatchedResourcePath();
-        String httpMethod = requestContext.getMatchedResourcePath().getMethod().toString();
         SignedJWTInfo signedJWTInfo;
         try {
             signedJWTInfo = getSignedJwt(jwtToken);
         } catch (ParseException | IllegalArgumentException e) {
-            throw new SecurityException("Not a JWT token. Failed to decode the token header.", e);
+            log.error("Failed to decode the token header", e);
+            throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
+                    APISecurityConstants.API_AUTH_GENERAL_ERROR,
```

Comment:
This is invalid token, We should send invalid token error

Reply:
added via https://github.com/wso2/product-microgateway/pull/1914/commits/4b567b0841950c3eba864b2a3ab89fe836c73a75